src/v/kafka/server/group.h

```diff
@@ -615,7 +619,7 @@ class group {
     ctx_log _ctxlog;
     ctx_log _ctx_txlog;
 
-    mutex _tx_mutex;
+    absl::flat_hash_map<model::producer_id, ss::lw_shared_ptr<mutex>> _tx_locks;
```

Comment:
I don't know why this is a `lw_shared_pt<mutex>` does it need to have shared lifetime?

Reply:
no particular reason other than my bias cultivated by years of Java and C#, I've changed it